packages/@react-spectrum/s2/src/SelectBoxGroup.tsx

```diff
@@ -0,0 +1,263 @@
+});
+
+export const SelectBoxGroupContext = createContext<ContextValue<Partial<SelectBoxGroupProps>, DOMRefValue<HTMLDivElement>>>(null);
+
+const gridStyles = style({
+  display: 'grid',
+  gridAutoRows: '1fr',
+  gap: {
+    gutterWidth: {
+      default: 16,
+      compact: 8,
+      spacious: 24
+    }
+  }
+}, getAllowedOverrides());
+
+
+// Selector Group component
+interface SelectorGroupProps {
+  allowMultiSelect: boolean,
+  children: ReactNode,
+  style?: React.CSSProperties,
+  className?: string,
+  onChange: (value: SelectBoxValueType) => void,
+  value?: SelectBoxValueType,
+  defaultValue?: SelectBoxValueType,
+  isRequired?: boolean,
+  isDisabled?: boolean,
+  label?: ReactNode
+}
+
+const SelectorGroup = forwardRef<HTMLDivElement, SelectorGroupProps>(function SelectorGroupComponent({
+  allowMultiSelect,
+  children,
+  className,
+  onChange,
+  value,
+  style,
+  defaultValue,
+  isRequired,
+  isDisabled,
+  label
+}, ref) {
+  const props = { 
+    isRequired,
+    isDisabled,
+    className,
+    style,
+    children,
+    onChange,
+    ref
+  };
+
+  return allowMultiSelect ? (
+    <AriaCheckboxGroup
+      {...props}
+      value={ensureArray(value)}
+      defaultValue={ensureArray(defaultValue)}
+      aria-label={label ? String(label) : undefined} />
+  ) : (
+    <AriaRadioGroup
+      {...props}
+      value={unwrapValue(value)}
+      defaultValue={unwrapValue(defaultValue)}
+      aria-label={label ? String(label) : undefined} />
+  );
+});
+
+/**
+ * SelectBox groups allow users to select one or more options from a list.
+ * All possible options are exposed up front for users to compare.
+ */
+export const SelectBoxGroup = /*#__PURE__*/ forwardRef(function SelectBoxGroup(props: SelectBoxGroupProps, ref: DOMRef<HTMLDivElement>) {
+  [props, ref] = useSpectrumContextProps(props, ref, SelectBoxGroupContext);
+  
+  let {
+    label,
+    children,
+    onSelectionChange,
+    defaultValue,
+    selectionMode = 'single',
+    size = 'M',
+    orientation = 'vertical',
+    isEmphasized,
+    numColumns = 2,
+    gutterWidth = 'default',
+    isRequired = false,
+    isDisabled = false,
+    UNSAFE_style
+  } = props;
+
+  const [value, setValue] = useState<SelectBoxValueType | undefined>(defaultValue);
+  const allowMultiSelect = selectionMode === 'multiple';
+  
+  const domRef = useDOMRef(ref);
+
+  const getChildrenToRender = () => {
+    const childrenToRender = React.Children.toArray(children).filter((x) => x);
+    try {
+      const childrenLength = childrenToRender.length;
+      if (childrenLength <= 0) {
+        throw new Error('Invalid content. SelectBox must have at least a title.');
+      }
+      if (childrenLength > 9) {
+        throw new Error('Invalid content. SelectBox cannot have more than 9 children.');
+      }
+    } catch (e) {
+      console.error(e);
+    }
+    return childrenToRender;
+  };
+
+  useEffect(() => {
+    if (value !== undefined) {
+      onSelectionChange(value);
```

Comment:
rather than call this in a useEffect, create a callback which set on `onChange`, and in that callback, set the value as well as calling `onSelectionChange`